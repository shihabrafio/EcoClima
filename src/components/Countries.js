const Countries = (props) => {
  const { cityName, cityAqi } = props;
  let status = null;
  let style = { fontWeight: '800' };

  if (cityAqi === 1) {
    status = 'Good';
    style = { ...style, color: '#0f0' };
  } else if (cityAqi === 2) {
    status = 'Fair';
    style = { ...style, color: '#5affa5' };
  } else if (cityAqi === 3) {
    status = 'Moderate';
    style = { ...style, color: '#ff0' };
  } else if (cityAqi === 4) {
    status = 'Poor';
    style = { ...style, color: '#fa0' };
  } else if (cityAqi === 5) {
    status = 'Very Poor';
    style = { ...style, color: '#f00' };
  }

  return (
    <div className="ctd-preview">
      <h2>{cityName}</h2>
      <div>
        <span>Air Quality: </span>
        <span style={style}>{status}</span>
      </div>
      <div>
        <span>Quality Index: </span>
        <span style={style}>{cityAqi}</span>
      </div>
    </div>
  );
};

export default Countries;
