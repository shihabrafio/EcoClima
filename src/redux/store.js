import { configureStore } from '@reduxjs/toolkit';
import CountryReducer from './countrySlice';

const store = configureStore({
  reducer: CountryReducer,
});

export default store;
