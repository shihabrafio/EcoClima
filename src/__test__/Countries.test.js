import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Countries from '../components/Countries';

const render = (component) => rtlRender(
  <Provider store={store}>
    {component}
  </Provider>,
);

describe('Countries', () => {
  it('render Countries Components', () => {
    render(<Countries />);
    expect(Countries).toMatchSnapshot();
  });
});
