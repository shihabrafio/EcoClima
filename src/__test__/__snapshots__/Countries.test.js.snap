// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` render Mission Components 1`] = `[Function]`;

exports[`Countires render Components 1`] = `[Function]`;

exports[`Countires render Countries Components 1`] = `[Function]`;

exports[`Countires render Mission Components 1`] = `[Function]`;

exports[`Countries render Countries Components 1`] = `[Function]`;

exports[`MyRocketss render Mission Components 1`] = `[Function]`;
